services/publishing/Tweek.Publishing.Service/Validation/NestedKeysValidator.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using LanguageExt;
+using Newtonsoft.Json;
+using Tweek.Publishing.Service.Model.Rules;
+
+namespace Tweek.Publishing.Service.Validation
+{
+    public class NestedKeysValidator : IValidator
+    {
+        public async Task Validate(string filePath, Func<string, Task<string>> reader)
+        {
+            var currentFilePath = Path.GetDirectoryName(filePath);
+            
+            while (currentFilePath != "manifests")
```

Comment:
Might be worth also checking for an empty string to avoid an endless loop in case we somehow don't have a path nested under `manifests` (maybe because of some future change)

Reply:
I think the pattern should cover it. (as it's not running for every file)

Reply:
But yeah, probably should throw if we get to empty string or "/" (not sure what's the behavior)

Reply:
An endless can be scary. Probably easy to unit test this case